src/__tests__/resources/asset.integration.spec.ts

```diff
@@ -0,0 +1,122 @@
+      name: 'test-grandchild',
+    };
+
+    test('parent', async () => {
+      const newRoot = {
+        ...newRootAsset,
+        externalId: 'test-root' + randomInt(),
+      };
+      const newChild = {
+        ...newChildAsset,
+        parentExternalId: newRoot.externalId,
+      };
+      const createdAssets = await client.assets.create([newRoot, newChild]);
+      expect(await createdAssets[1].parent()).toBeInstanceOf(Asset);
+      expect(await createdAssets[0].parent()).toBe(null);
+      await client.assets.delete([{ id: createdAssets[0].id }], {
+        recursive: true,
+      });
+    });
+
+    test('subtree', async () => {
+      const newRoot = {
+        ...newRootAsset,
+        externalId: 'test-root' + randomInt(),
+      };
+      const childArray = [];
+      for (let index = 0; index < 3; index++) {
+        childArray.push({
+          ...newChildAsset,
+          externalId: 'test-child' + randomInt() + index,
+          parentExternalId: newRoot.externalId,
+        });
+      }
+      const grandChildArray = [];
+      for (let index = 0; index < 105; index++) {
+        grandChildArray.push({
+          ...newGrandChildAsset,
+          parentExternalId: childArray[0].externalId,
+          externalId: 'test-grandchild' + randomInt() + index,
+        });
+      }
+      const createdAssets = await client.assets.create([
+        newRoot,
+        ...childArray,
+        ...grandChildArray,
+      ]);
+      await sleepPromise(5000); // eventual consistency in the backend
+      const subtreeWithDepth2 = await createdAssets[0].subtree({ depth: 2 });
+      const subtreeWithDepth1 = await createdAssets[0].subtree({ depth: 1 });
+      const subtreeWithoutSpecifiedDepth = await createdAssets[0].subtree();
+      expect(subtreeWithDepth2.length).toBe(109);
+      expect(subtreeWithDepth1.length).toBe(4);
+      expect(subtreeWithoutSpecifiedDepth.length).toBe(109);
+      await client.assets.delete([{ id: createdAssets[0].id }], {
+        recursive: true,
+      });
+    });
+
+    test('events from Asset', async () => {
+      await testResourceType(client, client.events, newRootAsset);
+    });
+
+    test('timeseries from Asset', async () => {
+      await testResourceType(client, client.timeseries, newRootAsset);
+    });
+  });
+});
+
+async function testResourceType(
+  client: CogniteClient,
+  api: TimeSeriesAPI | EventsAPI,
+  newRootAsset: any
+) {
+  const newRoot = {
+    ...newRootAsset,
+    externalId: 'test-root' + randomInt(),
+  };
+  const createdAssets = await client.assets.create([newRoot]);
+  const content =
+    api instanceof TimeSeriesAPI
+      ? { assetId: createdAssets[0].id }
+      : { assetIds: [createdAssets[0].id] };
+  const resourceList = new Array(1003).fill(content);
+  const resources = await api.create(resourceList);
+  await sleepPromise(10000); // eventual consistency in the backend
+  let fetchedResource: GetTimeSeriesMetadataDTO[] | CogniteEvent[];
```

Comment:
```suggestion
  let fetchedResource: (GetTimeSeriesMetadataDTO | CogniteEvent)[];
```
by doing this you can remove `// @ts-ignore` below

Reply:
Thanks!